pyproject.toml

```diff
@@ -37,6 +37,7 @@ pyyaml = "^5.3.1"
 arrow = "^0.15.5"
 pybase62 = "^0.4.0"
 dataclasses = {version = "^0.7", python = "~3.6"}
+pydantic = "^1.7.3"
```

Comment:
Another thing if we switch to the pydantic dataclass implementation, we can probably get rid of the dataclass import from the previous line of t his.

Reply:
Our ActionProviderDescription is the last thing that uses the dataclass decorator. For portability it may make sense to remove the use there and get rid of the package altogether? If we switch it over to the pydantic model type we get some field validation (which may be useful since we're expecting to interpret this on introspection) but otherwise dataclasses and pydantic models behave pretty similarly. I'm open to either direction.

im reminded of when id get tripped up updating things bc of the dataclass dependency, and that faint memory makes me want to remove it.